src/librustdoc/clean/auto_trait.rs

```diff
@@ -21,42 +21,41 @@ struct RegionDeps<'tcx> {
 }
 
 crate struct AutoTraitFinder<'a, 'tcx> {
-    crate cx: &'a core::DocContext<'tcx>,
+    crate cx: RefCell<&'a mut core::DocContext<'tcx>>,
```

Comment:
> Why is this use of RefCell being added?

Ugh, this whole bit was a mess. I couldn't figure out how to get it to work, this is the error without the RefCell:

```
error[E0500]: closure requires unique access to `self` but it is already borrowed
  --> src/librustdoc/clean/auto_trait.rs:56:82
   |
56 |                     self.f.find_auto_trait_generics(ty, param_env, trait_def_id, |infcx, info| {
   |                     ------ ------------------------                              ^^^^^^^^^^^^^ closure construction occurs here
   |                     |      |
   |                     |      first borrow later used by call
   |                     borrow occurs here
...
70 |                         let new_generics = self.param_env_to_generics(
   |                                            ---- second borrow occurs due to use of `self` in closure

```

> Also, if a cell is needed, can't we use Cell instead since references are Copy, or are references 'special'?

&T is Copy, but &mut T is not.

Reply:
OMG I'm a dumbass, I just needed to make `f` a separate field from the struct. Give me a second.

Reply:
Ok this no longer uses RefCell at all :tada: